### PR TITLE
add mco2 bonds

### PR DESCRIPTION
**Pull request by Atmosfearful**
Here's how new bonds are added @ladytrekker 

copy-pasta 😅 
Ctrl+shift+f `bct_usdc_lp` to find places where bonds are referenced
I know that it's an 18 decimal ERC20 same as `bct` so I just copy-paste what we do for `bct` (slightly different from `lp`).

Lots of room for cleanup in here going forward (reusing `ethers.Contract` instances, flatten some of the if statements, remove `let` mutations etc. But now is not the time 😎 

Had to tweak how units are displayed because previously it assumed we only value in BCT and SLP units. Now we have MCO2 units.

### CODE REVIEW

**Comment by Atmosfearful** [comment]
To streamline release and prevent bond snipers, this is going straight to prod once archi initializes the contracts and confirms it's working.

**Comment by ladytrekker** [comment]
> bct_usdc_lp

I guess thats the reason why the preview shows "loading" for the bond

**Review by ladytrekker — APPROVED** [review]
Thanks for the instructions. Looks straight forward

As mentioned below can you point me to some resources in this case for the mco2 bonds you added in this PR? Thanks

File: app/actions/bonds.ts [hunk 1]
```diff
@@ -45,6 +46,13 @@ export function contractForReserve(params: {
       params.provider
     );
   }
+  if (params.bond === "mco2") {
+    return new ethers.Contract(
+      addresses["mainnet"].mco2,
+      IERC20.abi,
+      params.provider
```
[the remark this answers is not in the document]
Comment (Atmosfearful) [reply app/actions/bonds.ts:53]: I dont have any internal docs on this stuff but the Ethers docs are pretty good. https://docs.ethers.io/v5/api/contract/contract/

File: app/actions/bonds.ts [hunk 2]
```diff
@@ -127,6 +135,10 @@ export const calcBondDetails = (params: {
       );
       bondQuote = await bondContract.payoutFor(valuation);
       bondQuote /= Math.pow(10, 9);
+    } else if (params.bond === "mco2") {
+      bondDiscount = (marketPrice * Math.pow(10, 9) - bondPrice) / bondPrice;
+      bondQuote = await bondContract.payoutFor(amountInWei);
+      bondQuote /= Math.pow(10, 18);
```
[the remark this answers is not in the document]
Comment (Atmosfearful) [reply app/actions/bonds.ts:141]: source is archi :P

[the remark this answers is not in the document]
Comment (kargakis) [reply app/actions/bonds.ts]: @ladytrekker the majority of ERC20 tokens have 18 decimals which means a human-readable price of 1 is quoted as 1 * 10^18 by the contracts. The bond contracts return prices with the same amount of decimals for consistency/sanity so we need to divide by 10^18 in UIs to present human-readable prices.

**Pull request merged**